components/lib/dialog/Dialog.js

```diff
@@ -47,6 +48,11 @@ export const Dialog = React.forwardRef((inProps, ref) => {
 
     useHandleStyle(DialogBase.css.styles, isUnstyled, { name: 'dialog' });
 
+    useOnEscape(maskRef, props.closable && props.closeOnEscape, (event) => {
+        onClose(event);
+        params.splice(paramLength - 1, 1);
```

Comment:
Why the `splice` here?  I don't understand what it does?

Reply:
@melloware Splice was part of the previous code when the escape action is done.

![image](https://github.com/primefaces/primereact/assets/123446355/4eb19240-2073-4d83-b2c1-06231e994fb8)

Reply:
event.stopImmediatePropogation is part of the new hook, so thats not needed here.

Reply:
But wont params.splice throw a null pointer exception because there are NO params here.  It was just a private variable in the old keyboard event?

Reply:
Resolved. Let me know your thoughts. Thanks,
@melloware